state/application_test.go

```diff
@@ -922,6 +922,240 @@ func (s *ApplicationSuite) TestUpdateConfigSettings(c *gc.C) {
 	}
 }
 
+func (s *ApplicationSuite) TestUpdateApplicationSeries(c *gc.C) {
+	ch := state.AddTestingCharmMultiSeries(c, s.State, "multi-series")
+	app := state.AddTestingServiceForSeries(c, s.State, "precise", "multi-series", ch)
+	err := app.UpdateApplicationSeries("trusty", false)
+	c.Assert(err, jc.ErrorIsNil)
+	assertApplicationSeriesUpdate(c, app, "trusty")
+}
+
+func (s *ApplicationSuite) TestUpdateApplicationSeriesSamesSeriesToStart(c *gc.C) {
+	ch := state.AddTestingCharmMultiSeries(c, s.State, "multi-series")
+	app := state.AddTestingServiceForSeries(c, s.State, "precise", "multi-series", ch)
+	err := app.UpdateApplicationSeries("precise", false)
+	c.Assert(err, gc.ErrorMatches, "no transaction operations are available")
```

Comment:
The "no transaction operations are available" error should not get exposed - the txn runner sees it and returns nil.
If this is required here then there's something wrong elsewhere.